runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/Scheduler.java

```diff
@@ -39,8 +40,13 @@
 @Service
 public class Scheduler {
   private static class FinishedWork {
+    Instant created = Instant.now();
     NotificationDto dto;
     int epoch;
+
+    public boolean shouldRerun() {
+      return !dto.getHealthType().isDone() && Duration.between(created, Instant.now()).toMinutes() < 10;
```

Comment:
Yes. This is necessary. When a directory is scanned, it's unclear how long it will be in the queue for. You can imagine a scenario like this:

t = 0: find a whole lot of directories (including X, which is running) and queue them
t = 14min: process directory X and write a finished work entry
t = 15min: scan directories again. Requeue X
t = 15 min: backlog clears, start reprocessing X